pkg/controllers/cassandra/seedlabeller/control.go

```diff
@@ -44,36 +44,60 @@ func NewControl(
 
 func (c *defaultSeedLabeller) labelSeedNodes(
 	cluster *v1alpha1.CassandraCluster,
+	np *v1alpha1.CassandraClusterNodePool,
 	set *appsv1beta1.StatefulSet,
 ) error {
-	// TODO: make number of seed nodes configurable
-	pod, err := c.pods.Pods(cluster.Namespace).Get(fmt.Sprintf("%s-%d", set.Name, 0))
-	if err != nil {
-		glog.Warningf("Couldn't get stateful set pod: %v", err)
-		return nil
-	}
-	labels := pod.Labels
-	value := labels[seedprovider.SeedLabelKey]
-	if value == seedprovider.SeedLabelValue {
-		return nil
-	}
-	if labels == nil {
-		labels = map[string]string{}
+	for i := int64(0); i < np.Replicas; i++ {
+		pod, err := c.pods.Pods(cluster.Namespace).Get(fmt.Sprintf("%s-%d", set.Name, i))
+		if err != nil {
+			glog.Warningf("Couldn't get stateful set pod: %v", err)
+			return nil
```

Comment:
:grey_question: I'm not sure about this. In other controllers we would return this error, but I can see why we'd want to continue and attempt to label other pods if one of the `Get`s fails.

Could we collect the errors and return an aggregate error at the end?

See `k8s.io/apimachinery/pkg/util/errors` `NewAggregate`